src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -716,6 +748,65 @@ static bool ReturnOperationBelongsToMethod(SyntaxNode returnOperationSyntax, Syn
 
                     return enclosingMethod == methodSyntax;
                 }
+
+                async Task<SemanticDocument> CheckReturnOperations(SyntaxNode node)
```

Comment:
This should probably be `async Task<bool> TryUpdateReturnTypeForNullability(SyntaxNode node, out SemanticDocument updatedDocument)`. Then you can use it like `if (await TryUpdateReturnTypeForNullability(localMethodDeclaration, out var updatedDocument).ConfigureAwait(false))` at the callsites

Reply:
I tried this but was met with the error that async methods can't have out variables. 😞